### ISSUE TITLE
Client IP Spoof

### ISSUE THREAD

**Issue by sonofwill** [yyyar/gobetween#230]
Hello,

I would like the backend server to see the source IP of the client.

Can this be done? I have tried the transparent mode. But with this mode enabled, the backend still only sees the IP of the go-between Load balancer.

**Comment by zerodayyy** [comment]
That's not how an L4 load balancer works. If you spoof the source IP, the response will go directly to client from your node itself, which will mess with sessions, as the response will be sent from different IP address than the request was sent to. In your case, you need L3 balancing, i.e. a router with anycast IPs on your nodes.

**Comment by sonofwill** [comment]
I understand this. But it is easy to redirect the packets to return to the
client via the load balancer.

**Comment by zerodayyy** [comment]
That would only be possible if you configure LB as the default gateway on
your node. Again, this requires an L3 balancer, which gobetween is not.

**Comment by nuttysquirrel** [comment]
it feels like you're thinking with a L4/TCP hat. 
UDP can work this way.

**Comment by sonofwill** [comment]
I have it working using another UDP load balancer.

When the UDP packets return to the client via LB, I use iptables to send it
to localhost to have it handled by the application.

**Comment by nuttysquirrel** [comment]
yep, sorry - not directed at you. clarified my original comment :)
i'm playing with udp stuff too... what are you using?

**Comment by sonofwill** [comment]
Nginx.
